### PR TITLE
Converted RNTesterReactHostDelegate.java to RNTesterReactHostDelegate.kt

### PR DESCRIPTION
**Pull request by fabiansolheim**
## Summary:

This PR converts RNTesterReactHostDelegate into Kotlin as requested in https://github.com/facebook/react-native/issues/38825 .

## Changelog:

[INTERNAL] [CHANGED] - Convert to RNTesterReactHostDelegate.java => Kotlin

## Test Plan:

Run yarn && yarn android
All tests should pass

### CODE REVIEW

**Comment by fabiansolheim** [comment]
messed up my rebase, new branch it is😅 @cortinico

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/RNTesterReactHostDelegate.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import com.facebook.react.shell.MainReactPackage
+import com.facebook.react.turbomodule.core.TurboModuleManager
+import com.facebook.react.uiapp.component.MyLegacyViewManager
+import com.facebook.react.uiapp.component.MyNativeViewManager
+import com.facebook.react.uimanager.ViewManager
+
+@UnstableReactNativeAPI
+class RNTesterReactHostDelegate internal constructor(context: Context) : ReactHostDelegate {
+  var reactHost: ReactHostImpl? = null
+  private var mReactPackages: List<ReactPackage> = emptyList()
```
Comment (cortinico) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/RNTesterReactHostDelegate.kt:39]:
```suggestion
  private var reactPackages: List<ReactPackage> = emptyList()
```

**Review by cortinico — CHANGES_REQUESTED** [review]
Thanks for the follow-up. There is one small nit to update and then we can import/merge it 👍

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/RNTesterReactHostDelegate.kt (hunk 1 above)
Comment (fabiansolheim) [reply packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/RNTesterReactHostDelegate.kt:39]: @cortinico Alright awesome! But this change would clash with the existing property already declared on line 63, would you suggest a new name for that?

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/RNTesterReactHostDelegate.kt [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+              reactContext: ReactApplicationContext
+            ): List<NativeModule> = emptyList()
+
+            override fun createViewManagers(
+              reactContext: ReactApplicationContext
+            ): List<ViewManager<*, *>> =
+              listOf(MyNativeViewManager(), MyLegacyViewManager(reactContext))
+          }
+        )
+    reactPackages
```
Comment (cortinico) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/RNTesterReactHostDelegate.kt:115]:
```suggestion
```

**Review by cortinico — APPROVED** [review]

**Comment by cortinico** [comment]
@fabiansolheim could you rebase?

**Pull request closed without merging**